app/assets/javascripts/reducers/uploads.js

```diff
@@ -31,6 +34,35 @@ export default function uploads(state = initialState, action) {
         ...state,
         uploads: sortedModel.newModels,
         sortKey: sortedModel.newKey,
+        fetchState: false,
+      };
+    }
+    case SET_UPLOAD_METADATA: {
+      let fetchedData;
+      _.forEach(action.data, data => {
+        if (data.query) {
+          fetchedData = { ...fetchedData, ...data.query.pages };
+        }
+      });
+      const updatedUploads = state.uploads.map(upload => {
+        if (fetchedData && fetchedData[upload.id]) {
+          if (fetchedData[upload.id].imageinfo[0].extmetadata.Credit) {
+            upload.credit = fetchedData[upload.id].imageinfo[0].extmetadata.Credit.value;
+          }
+          if (!fetchedData[upload.id].imageinfo[0].extmetadata.Credit) {
```

Comment:
This would be clearer and less repetitive as an `else`, since we'll always either find a credit or set the fallback.

Reply:
I replaced this with lodash's  _.get to avoid checking for every key if it exists